ui/components/component-library/avatar-base/avatar-base.types.ts

```diff
@@ -45,3 +46,10 @@ export interface AvatarBaseProps extends TextProps {
    */
   className?: string;
 }
+
+export type AvatarBaseProps<C extends React.ElementType> =
+  PolymorphicComponentPropWithRef<C, Props>;
+
+export type AvatarBaseComponent = <C extends React.ElementType = 'span'>(
+  props: AvatarBaseProps<C>,
+) => React.ReactElement | null;
```

Comment:
Same here see if we can remove these updates and purely focus on the `Text` component

Reply:
Turns out we do need these because we are updating the `TextProps`